addon/search/search.js

```diff
@@ -158,6 +201,10 @@
     });
   }
 
+  CodeMirror.defineOption("incrementalSearch", false, function(cm, val) {
+    cm.state.incrementalSearch = (val === true);
```

Comment:
I'd prefer this as a separate command that passes around its own state. It is conceivable to want both incremental and non-incremental search on the same instance, so having it as an editor-global flag is undesirable.